bin/saya/src/args/shard.rs

```diff
@@ -0,0 +1,69 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use clap::builder::PossibleValue;
+use clap::{Args, ValueEnum};
+use katana_primitives::FieldElement;
+use saya_core::SayaMode;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct SayaModeArg(pub SayaMode);
+
+#[derive(Debug, Args, Clone)]
+pub struct ShardOptions {
```

Comment:
We should think how the terminology will be separated, as `shard` is mostly for ephemeral execution and persistent execution is done in a different manner at some point.
Or the name shouldn't be `piltover` but `settlement_contract` which can be the world for shards and piltover for persistency.

Reply:
Sure, that makes sense.